tests/indexes/test_datetime_index.py

```diff
@@ -0,0 +1,118 @@
+
+    dt_idx = pd.DatetimeIndex(data, name="date")
+    check(assert_type(data[x <= dt_idx], pd.DatetimeIndex), pd.DatetimeIndex)
+    check(assert_type(data[x >= dt_idx], pd.DatetimeIndex), pd.DatetimeIndex)
+    check(assert_type(data[x < dt_idx], pd.DatetimeIndex), pd.DatetimeIndex)
+    check(assert_type(data[x > dt_idx], pd.DatetimeIndex), pd.DatetimeIndex)
+    check(assert_type(data[dt_idx <= x], pd.DatetimeIndex), pd.DatetimeIndex)
+    check(assert_type(data[dt_idx >= x], pd.DatetimeIndex), pd.DatetimeIndex)
+    check(assert_type(data[dt_idx < x], pd.DatetimeIndex), pd.DatetimeIndex)
+    check(assert_type(data[dt_idx > x], pd.DatetimeIndex), pd.DatetimeIndex)
+
+    ind = pd.Index([1, 2, 3])
+    check(assert_type(ind <= 2, np_1darray[np.bool]), np_1darray[np.bool])
+    check(assert_type(ind >= 2, np_1darray[np.bool]), np_1darray[np.bool])
+    check(assert_type(ind < 2, np_1darray[np.bool]), np_1darray[np.bool])
+    check(assert_type(ind > 2, np_1darray[np.bool]), np_1darray[np.bool])
+
+
+def test_datetime_index_constructor() -> None:
+    check(assert_type(pd.DatetimeIndex(["2020"]), pd.DatetimeIndex), pd.DatetimeIndex)
+    check(
+        assert_type(pd.DatetimeIndex(["2020"], name="ts"), pd.DatetimeIndex),
+        pd.DatetimeIndex,
+    )
+    check(
+        assert_type(pd.DatetimeIndex(["2020"], freq="D"), pd.DatetimeIndex),
+        pd.DatetimeIndex,
+    )
+    check(
+        assert_type(pd.DatetimeIndex(["2020"], tz="Asia/Kathmandu"), pd.DatetimeIndex),
+        pd.DatetimeIndex,
+    )
+
+    # https://github.com/microsoft/python-type-stubs/issues/115
+    df = pd.DataFrame({"A": [1, 2, 3], "B": [5, 6, 7]})
+
+    check(
+        assert_type(
+            pd.DatetimeIndex(data=df["A"], tz=None, ambiguous="NaT", copy=True),
+            pd.DatetimeIndex,
+        ),
+        pd.DatetimeIndex,
+    )
+
+
+def test_intersection() -> None:
+    # GH 744
+    index = pd.DatetimeIndex(["2022-01-01"])
+    check(assert_type(index.intersection(index), pd.DatetimeIndex), pd.DatetimeIndex)
+    check(
+        assert_type(index.intersection([pd.Timestamp("1/1/2023")]), pd.DatetimeIndex),
+        pd.DatetimeIndex,
+    )
+
+
+def test_datetime_index_max_min_reductions() -> None:
+    dtidx = pd.DatetimeIndex(["2020-01-01", "2020-01-02"])
+    check(assert_type(dtidx.argmax(), np.int64), np.int64)
+    check(assert_type(dtidx.argmin(), np.int64), np.int64)
+    check(assert_type(dtidx.max(), pd.Timestamp), pd.Timestamp)
+    check(assert_type(dtidx.min(), pd.Timestamp), pd.Timestamp)
+
+
+def test_datetimeindex_shift() -> None:
+    ind = pd.date_range("2023-01-01", "2023-02-01")
+    check(assert_type(ind.shift(1), pd.DatetimeIndex), pd.DatetimeIndex)
+
+
+def test_datetimeindex_indexer_at_time() -> None:
+    dti = pd.date_range("2023-01-01", "2023-02-01")
+    check(assert_type(dti.indexer_at_time("10:00"), np_1darray[np.intp]), np_1darray)
+    check(assert_type(dti.indexer_at_time(time(10)), np_1darray[np.intp]), np_1darray)
```

Comment:
Would the following work?
```suggestion
    check(assert_type(dti.indexer_at_time("10:00"), np_1darray[np.intp]), np_1darray[np.intp], np.intp)
    check(assert_type(dti.indexer_at_time(time(10)), np_1darray[np.intp]), np_1darray[np.intp], np.intp)
```